sdk/go/message_test.go

```diff
@@ -0,0 +1,151 @@
+//
+// Copyright (c) 2018
+// Mainflux
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package sdk_test
+
+import (
+	"fmt"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/mainflux/mainflux"
+	adapter "github.com/mainflux/mainflux/http"
+	"github.com/mainflux/mainflux/http/api"
+	"github.com/mainflux/mainflux/http/mocks"
+	sdk "github.com/mainflux/mainflux/sdk/go"
+	"github.com/stretchr/testify/assert"
+)
+
+func newMessageService() mainflux.MessagePublisher {
+	pub := mocks.NewPublisher()
+	return adapter.New(pub)
+}
+
+func newMessageServer(pub mainflux.MessagePublisher, cc mainflux.ThingsServiceClient) *httptest.Server {
+	mux := api.MakeHandler(pub, cc)
+	return httptest.NewServer(mux)
+}
+
+func TestSendMessage(t *testing.T) {
+	chanID := "1"
+	invalidID := "wrong"
+	atoken := "auth_token"
+	invalidToken := "invalid_token"
+	msg := `[{"n":"current","t":-1,"v":1.6}]`
+	id, _ := strconv.ParseUint(chanID, 10, 64)
+
+	thingsClient := mocks.NewThingsClient(map[string]uint64{atoken: id})
+	fmt.Println(thingsClient)
+	pub := newMessageService()
+	ts := newMessageServer(pub, thingsClient)
+	defer ts.Close()
+	sdkConf := sdk.Config{
+		BaseURL:           ts.URL,
+		UsersPrefix:       "",
+		ThingsPrefix:      "",
+		HTTPAdapterPrefix: "",
+		MsgContentType:    contentType,
+		TLSVerification:   false,
+	}
+
+	mainfluxSDK := sdk.NewSDK(sdkConf)
+
+	cases := map[string]struct {
+		chanID string
+		msg    string
+		auth   string
+		err    error
+	}{
+		"publish message": {
+			chanID: chanID,
+			msg:    msg,
+			auth:   atoken,
+			err:    nil,
+		},
+		"publish message without authorization token": {
+			chanID: chanID,
+			msg:    msg,
+			auth:   "",
+			err:    sdk.ErrUnauthorized,
+		},
+		"publish message with invalid authorization token": {
+			chanID: chanID,
+			msg:    msg,
+			auth:   invalidToken,
+			err:    sdk.ErrUnauthorized,
+		},
+		"publish message with wrong content type": {
+			chanID: chanID,
+			msg:    "text",
+			auth:   atoken,
+			err:    nil,
+		},
+		"publish message to wrong channel": {
+			chanID: invalidID,
+			msg:    msg,
+			auth:   atoken,
+			err:    sdk.ErrNotFound,
+		},
+		"publish message unable to authorize": {
+			chanID: chanID,
+			msg:    msg,
+			auth:   mocks.ServiceErrToken,
+			err:    sdk.ErrFailedUpdate,
```

Comment:
Add another error to SDK (`ErrFailedPublish`) and replace this error with that one [here](https://github.com/mainflux/mainflux/blob/master/sdk/go/message.go#L39).